lib/plugins/cleaner.rb

```diff
@@ -0,0 +1,51 @@
+require 'rubygems/package'
+require_relative '../moonshot/creds_helper'
+
+module Moonshot
+  module Plugins
+    # Moonshot plugin class for removing artifacts when a stage gets deleted.
+    class Cleaner
+      include Moonshot::CredsHelper
+
+      def initialize(bucket_name)
+        @bucket_name = bucket_name
+      end
+
+      def post_delete(resources)
+        stack_name = resources.stack.name
+        resources.ilog.start 'Starting to delete artifacts for stack: ' \
+          "#{stack_name}" do |s|
+          arts = retrieve_artifact_names(stack_name)
+          if arts.empty?
+            s.success 'No artifacts found, nothing deleted.'
+          else
+            delete_artifacts arts
+            s.success "Deleted artifacts for stack: '#{stack_name}' " \
+                      "in bucket: '#{@bucket_name}'"
+          end
+        end
+      end
+
+      private
+
+      def retrieve_artifact_names(name)
+        objs = s3_client.list_objects(bucket: @bucket_name, prefix: name)
+        valid_keys = []
+        objs.contents.each do |o|
+          valid_keys << o.key if o.key =~ /^#{name}[\.-]/
```

Comment:
Looking at http://docs.aws.amazon.com/codedeploy/latest/userguide/limits.html it looks like we should accept .tar.gz, .tar, and .zip as those are the only ones allowed for use with code deploy. Unless we want this plugin to be more generic and not tied specifically to cleaning up only code deploy artifacts.